### PR TITLE
Adding Maintenance Window & External Links Data sources

### PR DESCRIPTION
**Pull request by dkuldeep22**

### CODE REVIEW

**Review by yogeshprasad — APPROVED** [review]

File: docs/data-sources/external_link.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+Use this data source to get information about a Wavefront external link by its ID.
+
+## Argument Reference
+* `id` - (Required) The ID of the external link.
+
+## Example Usage
+
+```hcl
+# Get the information about a specific external links.
```
Comment (Margarita-Staneva) [inline docs/data-sources/external_link.md:18]: # Get the information about a specific external link.

File: docs/data-sources/external_link.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+data "wavefront_external_link" "example" {
+  id = "sample-external-link-id"
+}
+```
+
+## Attribute Reference
+
+* `name` - The name of the external link.
+* `id` -  The ID of the external link.
+* `description` - Human-readable description for this link.
```
Comment (Margarita-Staneva) [inline docs/data-sources/external_link.md:28]: Human-readable description of this link.

File: docs/data-sources/external_links.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+layout: "wavefront"
+page_title: "Wavefront: External Links"
+description: |-
+    Get the information about all Wavefront External Links.
```
Comment (Margarita-Staneva) [inline docs/data-sources/external_links.md:5]: Get the information about all Wavefront external links.

File: docs/data-sources/external_links.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  offset = 0
+}
+```
+
+## Attribute Reference
+
+* `external_links` - List of all external links in Wavefront. For each external link you will see a list of attributes.
+  * `name` - The name of the external link.
+  * `id` -  The ID of the external link.
+  * `description` - Human-readable description for this link.
```
Comment (Margarita-Staneva) [inline docs/data-sources/external_links.md:31]: Human-readable description of the link.

File: docs/data-sources/external_links.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  * `template` - The mustache template for this link. The template must expand to a full URL, including scheme, origin, etc.
```
Comment (Margarita-Staneva) [inline docs/data-sources/external_links.md:32]: The mustache template for the link. The template must expand to a full URL, including scheme, origin, etc.

File: docs/data-sources/maintenance_window.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+Use this data source to get information about a Wavefront maintenance window by its ID.
+
+## Argument Reference
+* `id` - (Required) The ID of the maintenance window.
+
+## Example Usage
+
+```hcl
+# Get the information about specific maintenance window.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:18]: Get the information about a specific maintenance window.

File: docs/data-sources/maintenance_window.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+* `reason` - The reason for the maintenance window.
+* `title` - The title of the maintenance window.
+* `customer_id` - The ID of the customer in Wavefront.
+* `running_state` - The running state of the maintenance window.
+* `start_time_in_seconds` - The start time in seconds after 1 Jan 1970 GMT.
+* `end_time_in_seconds` - The end time in seconds after 1 Jan 1970 GMT.
+* `relevant_customer_tags` - The list of alert tags whose matching alerts will be put into maintenance because
+  of this maintenance window. At least one of `relevant_customer_tags`, `relevant_host_tags`, or `relevant_host_names`
+  is required.
+* `relevant_host_tags` - The list of source/host tags whose matching sources/hosts will be put into maintenance
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:37]: The list of source or host tags whose matching sources or hosts will be put into maintenance

File: docs/data-sources/maintenance_window.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  because of this maintenance window. At least one of `relevant_customer_tags`, `relevant_host_tags`, or
+  `relevant_host_names` is required.
+* `relevant_host_names` - The list of source/host names that will be put into maintenance because of this
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:40]: The list of source or host names...

File: docs/data-sources/maintenance_window.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  maintenance window. At least one of `relevant_customer_tags`, `relevant_host_tags`, or `relevant_host_names`
+  is required.
+* `relevant_host_tags_anded` - Whether to AND source/host tags listed in `relevant_host_tags`.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:43]: Whether to AND the source or host tags listed in...

File: docs/data-sources/maintenance_window.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  If `true`, a source/host must contain all tags in order for the maintenance window to apply. If `false`,
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:44]: If set to `true`, the source or host must contain all tags for the maintenance window to apply. If set to `false`,
  the tags are OR'ed, and the source or host must contain one of the tags. Default value is `false`.

File: docs/data-sources/maintenance_window.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+  the tags are OR'ed, and a source/host must contain one of the tags. Default: `false`.
+* `host_tag_group_host_names_group_anded` - If `true`, a source/host must be in `relevant_host_names`
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:46]: If set to `true`, the source or host must be in `relevant_host_names` and must have tags matching the specification formed by `relevant_host_tags` and `relevant_host_tags_anded` in for this maintenance window to apply. If  set to `false`, the source or host must either be in `relevant_host_names` or match `relevant_host_tags` and `relevant_host_tags_anded`. Default value is `false`.

File: docs/data-sources/maintenance_window.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  and have tags matching the specification formed by `relevant_host_tags` and `relevant_host_tags_anded` in
+  order for this maintenance window to apply. If `false`, a source/host must either be in `relevant_host_names`
+  or match `relevant_host_tags` and `relevant_host_tags_anded`. Default: `false`.
+* `created_epoch_millis` - The timestamp in epoch milliseconds indicating when the external link is created.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:50]: The timestamp in epoch milliseconds indicating when the maintenance window is created.

File: docs/data-sources/maintenance_window.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+* `updated_epoch_millis` - The timestamp in epoch milliseconds indicating when the external link is updated.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:51]: The timestamp in epoch milliseconds indicating when the maintenance window is updated.

File: docs/data-sources/maintenance_window.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+* `creator_id` - The ID of the user who created the external link.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:52]: The ID of the user who created the maintenance window.

File: docs/data-sources/maintenance_window.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+* `updater_id` - The ID of the user who updated the external link.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_window.md:53]: The ID of the user who updated the maintenance window.

File: docs/data-sources/maintenance_windows.md [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+Use this data source to get information about all Wavefront maintenance windows.
+
+## Argument Reference
+* `limit` - (Optional) Limit is the maximum number of results to be returned. Defaults to 100.
+* `offset` - (Optional) Offset is the offset from the first result to be returned. Defaults to 0.
+
+## Example Usage
+
+```hcl
+# Get the information about all external links.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_windows.md:19]: Get the information about all maintenance windows.

File: docs/data-sources/maintenance_windows.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+data "wavefront_maintenance_window_all" "example" {
+  limit = 10
+  offset = 0
+}
+```
+
+## Attribute Reference
+
+* `maintenance_windows` - List of all external links in Wavefront. For each external link you will see a list of attributes.
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_windows.md:28]: List of all maintenance windows in Wavefront. For each maintenance window you will see a list of attributes.

File: docs/data-sources/maintenance_windows.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    * `reason` - The reason for the maintenance window.
+    * `title` - The title of the maintenance window.
+    * `customer_id` - The ID of the customer in Wavefront.
+    * `running_state` - The running state of the maintenance window.
+    * `start_time_in_seconds` - The start time in seconds after 1 Jan 1970 GMT.
+    * `end_time_in_seconds` - The end time in seconds after 1 Jan 1970 GMT.
+    * `relevant_customer_tags` - The list of alert tags whose matching alerts will be put into maintenance because
+      of this maintenance window. At least one of `relevant_customer_tags`, `relevant_host_tags`, or `relevant_host_names`
+      is required.
+    * `relevant_host_tags` - The list of source/host tags whose matching sources/hosts will be put into maintenance
```
Comment (Margarita-Staneva) [inline docs/data-sources/maintenance_windows.md:40]: Please apply the comments from the review of maintenance_window.md.

**Review by Margarita-Staneva — CHANGES_REQUESTED** [review]
I've got some suggestions for the text. 

Thanks!

File: docs/data-sources/maintenance_windows.md (hunk 18 above)
Comment (dkuldeep22) [reply docs/data-sources/maintenance_windows.md]: Done! Thank you @Margarita-Staneva !

**Pull request merged**